### PR TITLE
add dev-env tests

### PR DESCRIPTION
**Pull request by alberhander**
- Adding dev-env tests on pr builds.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -60,6 +61,51 @@ jobs:
...
+        docker stop postgres redis
+
+    # dev-env tests
+    - uses: actions/checkout@v2
+      with:
+        repository: 'cartodb/docker-dev-env'
+        path: 'docker-dev-env'
+        token: ${{ secrets.CARTOFANTE_PERSONAL_TOKEN }}
+
+    - name: Check docker-compose syntax
```
Comment (ilbambino) [inline .github/workflows/main.yml:75]: Do you think this step adds value? In `dev-env` where the compose file changes I agree, but here?

**Review by ilbambino — COMMENTED** [review]
would it make sense to do it as another job? 
if it fails it would be easier to know why. 
I don't know if we could use the image from one job and reuse it in others (I haven't read the docs if possible).
Also if we add later (soon) tests for onprem they could run in parallel and if they are separated jobs the failures would be clearer

File: .github/workflows/main.yml (hunk 1 above)
Comment (alberhander) [reply .github/workflows/main.yml:75]: I just left it for debugging purposes, because the docker-compose file lives in a different repository.

**Comment by alberhander** [comment]
> would it make sense to do it as another job?
> if it fails it would be easier to know why.
> Once the image is built it could be [cached](https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows) and reused in the other jobs. (add a dep using `needs: build-and-test`)
> Also if we add later (soon) tests for onprem they could run in parallel and if they are separated jobs the failures would be clearer

I didn't add a different job because most of the repos are in Cloud Build and there's no such option there, but it makes sense doing it here, it will be easier to troubleshoot any failure. I'll make them work (I'm missing gcr permissions) and I'll run them in a separate job!

**Pull request merged**